tests/benchq/resource_estimation/test_extrapolation_estimation.py

```diff
@@ -0,0 +1,268 @@
+            "logarithmic",
+        ),
+        (
+            QuantumProgram(
+                [
+                    Circuit([H(0), RZ(np.pi / 14)(0), CNOT(0, 1)]),
+                    Circuit([H(0), H(1), RZ(np.pi / 14)(0)]),
+                ],
+                1000,
+                lambda x: [0] + [1] * x + [0],
+            ),
+            [1, 2, 3, 5],
+            "logarithmic",
+        ),
+        (
+            QuantumProgram(
+                [
+                    Circuit([H(0), RZ(np.pi / 14)(0), CNOT(0, 1)]),
+                    Circuit([H(0), H(1), RZ(np.pi / 14)(0)]),
+                    Circuit([H(0), H(1), CNOT(0, 1)]),
+                ],
+                100,
+                lambda x: [0] + [1, 2] * x + [0],
+            ),
+            [1, 2, 3, 5, 7, 10, 15, 25],
+            "linear",
+        ),
+    ],
+)
+def test_get_resource_estimations_for_program_gives_correct_results(
+    quantum_program,
+    steps_to_extrapolate_from,
+    use_delayed_gate_synthesis,
+    n_measurement_steps_fit_type,
+):
+    architecture_model = BasicArchitectureModel(
+        physical_gate_error_rate=1e-3,
+        physical_gate_time_in_seconds=1e-6,
+    )
+    error_budget = {
+        "qsp_required_precision": 1e-3,
+        "tolerable_circuit_error_rate": 1e-2,
+        "total_error": 1e-2,
+        "synthesis_error_rate": 0.5,
+        "ec_error_rate": 0.5,
+    }
+    transformers = _get_transformers(use_delayed_gate_synthesis, error_budget)
+
+    extrapolated_resource_estimates = run_extrapolation_pipeline(
+        quantum_program,
+        error_budget,
+        estimator=ExtrapolationResourceEstimator(
+            architecture_model,
+            steps_to_extrapolate_from,
+            n_measurement_steps_fit_type=n_measurement_steps_fit_type,
+        ),
+        transformers=transformers,
+    )
+    gsc_resource_estimates = run_resource_estimation_pipeline(
+        quantum_program,
+        error_budget,
+        estimator=GraphResourceEstimator(architecture_model),
+        transformers=transformers,
+    )
+
+    attributes_to_compare_harshly = [
+        "n_nodes",
+        "synthesis_multiplier",
+        "code_distance",
+        "total_time",
+        "n_physical_qubits",
+    ]
+    for attribute in attributes_to_compare_harshly:
+        assert np.isclose(
+            getattr(extrapolated_resource_estimates, attribute),
+            getattr(gsc_resource_estimates, attribute),
+            rtol=1e-2,
+        )
+    # assert that the number of measurement steps grows with the steps
+    attributes_to_compare_loosely = [
+        "n_logical_qubits",
+        "n_measurement_steps",
+    ]
+    for attribute in attributes_to_compare_loosely:
+        assert (
+            getattr(extrapolated_resource_estimates, attribute)
+            - getattr(
+                extrapolated_resource_estimates.data_used_to_extrapolate[-1], attribute
+            )
+            > -3  # allow for fits that might be a bit off
```

Comment:
This assumes that your fit always underestimates – should we use abs on the difference here?

Reply:
No, it just allows for small underestimates for smaller cases. Which can happen if `n_measurement_steps` is growing slowly.